stores/viewer.js

```diff
@@ -15,13 +14,29 @@ export const use_viewer_store = defineStore("viewer", {
     request_counter: 0,
   }),
   getters: {
-    base_url(state) {
+    protocol() {
+      if (use_infra_store().is_cloud) {
+        return "wss"
+      } else {
+        return "ws"
+      }
+    },
+    port() {
+      if (use_infra_store().is_cloud) {
+        return "443"
+      } else {
+        return this.default_local_port
+      }
+    },
+    base_url() {
       const infra_store = use_infra_store()
-      var viewer_url = `${state.PROTOCOL}://${infra_store.domain_name}:${state.PORT}`
+      var viewer_url = `${this.protocol}://${infra_store.domain_name}:${this.port}`
```

Comment:
Let or var?
this.port or port()? Same question in the other store and for protocol

Reply:
Indeed for var, getters aren't functions, so no parenthesis